### PR TITLE
Register validator beacon api

### LINKED ISSUE #11220: Add `/eth/v1/validator/register_validator` to Beacon API
# 🚀 Feature Request

### Description

Prysm can `register_validator` via gRPC, but not yet via the Beacon API.

This would be great for 3rd party validator clients such as [Vouch](github.com/attestantio/vouch).

[Link to spec](https://ethereum.github.io/beacon-APIs/#/Validator/registerValidator) <--

🙏

### PR DESCRIPTION
**Pull request by james-prysm**
**What type of PR is this?**

Missed Feature

**What does this PR do? Why is it needed?**
This PR adds the HTTP REST Beacon API endpoint ( [spec list here ](https://ethereum.github.io/beacon-APIs/#/Validator/registerValidator)) for the validator registration API endpoint.

**Which issues(s) does this PR fix?**

Fixes #11220

### CODE REVIEW

File: beacon-chain/rpc/eth/validator/validator.go [hunk 1]
```diff
@@ -579,6 +582,38 @@ func (vs *Server) PrepareBeaconProposer(
 	return &emptypb.Empty{}, nil
 }
 
+// SubmitValidatorRegistration submits validator registrations.
+func (vs *Server) SubmitValidatorRegistration(ctx context.Context, reg *ethpbv1.SubmitValidatorRegistrationsRequest) (*empty.Empty, error) {
+	ctx, span := trace.StartSpan(ctx, "validator.ValidatorRegistration")
+	defer span.End()
+
+	if vs.V1Alpha1Server.BlockBuilder == nil || !vs.V1Alpha1Server.BlockBuilder.Configured() {
+		return &empty.Empty{}, status.Errorf(codes.InvalidArgument, "Could not register block builder: %v", builder.ErrNoBuilder)
```
Comment (terencechain) [inline beacon-chain/rpc/eth/validator/validator.go:591]:
```suggestion
		return &empty.Empty{}, status.Errorf(codes.Internal, "Could not register block builder: %v", builder.ErrNoBuilder)
```

File: beacon-chain/rpc/eth/validator/validator.go [hunk 2]
```diff
@@ -8,6 +8,8 @@ import (
 	"strconv"
 	"time"
 
+	"github.com/prysmaticlabs/prysm/beacon-chain/builder"
+
```
Comment (rkapka) [inline beacon-chain/rpc/eth/validator/validator.go:12]:
```suggestion
```

File: beacon-chain/rpc/eth/validator/validator.go [hunk 3]
```diff
@@ -579,6 +582,38 @@ func (vs *Server) PrepareBeaconProposer(
 	return &emptypb.Empty{}, nil
 }
 
+// SubmitValidatorRegistration submits validator registrations.
+func (vs *Server) SubmitValidatorRegistration(ctx context.Context, reg *ethpbv1.SubmitValidatorRegistrationsRequest) (*empty.Empty, error) {
+	ctx, span := trace.StartSpan(ctx, "validator.ValidatorRegistration")
```
Comment (rkapka) [inline beacon-chain/rpc/eth/validator/validator.go:587]:
```suggestion
	ctx, span := trace.StartSpan(ctx, "validator.SubmitValidatorRegistration")
```

File: beacon-chain/rpc/eth/validator/validator.go [hunk 4]
```diff
@@ -579,6 +582,38 @@ func (vs *Server) PrepareBeaconProposer(
...
+				FeeRecipient: message.FeeRecipient,
+				GasLimit:     message.GasLimit,
+				Timestamp:    message.Timestamp,
+				Pubkey:       message.Pubkey,
+			},
+			Signature: registration.Signature,
+		})
+	}
+	if len(registrations) == 0 {
+		return &empty.Empty{}, status.Errorf(codes.InvalidArgument, "validator registration request is empty")
```
Comment (rkapka) [inline beacon-chain/rpc/eth/validator/validator.go:607]:
```suggestion
		return &empty.Empty{}, status.Errorf(codes.InvalidArgument, "Validator registration request is empty")
```

**Review by rkapka — CHANGES_REQUESTED** [review]
This also needs API Middleware setup

**Review by rkapka — APPROVED** [review]

**Pull request merged**